documentation/python.py

```diff
@@ -907,47 +907,59 @@ def enclosing_module_for(state: State, name: str) -> Optional[str]:
     return None
 
 def add_module_dependency_for(state: State, object: Union[Any, str]):
+    """Update the set of module dependencies for `state.current_module`.
+
+    Based on the type of `object`, add the module it's defined in to the set of
+    dependencies for the current module. If the object is a string, it's assumed
+    to be a type name that's already in the name map and the module name is
+    extracted from it. If the object is a module, its name is extracted from
+    `object.__name__`. If the object is a class/function/enum/..., its module
+    name is extracted from `object.__module__`.
+    """
+
     assert state.current_module
 
     # If not a string and not a module, try looking if its name-mapped type is
     # in our name map. If it is, and its enclosing module is as well, use that
     # to have name mapping correctly applied for it.
-    name = None
+    module_name = None
     if not isinstance(object, str) and not inspect.ismodule(object):
         name_candidate = map_name_prefix(state, extract_type(object))
         if name_candidate in state.name_map:
-            name = enclosing_module_for(state, name_candidate)
+            module_name = enclosing_module_for(state, name_candidate)
 
     # If the above didn't succeed, try other options
-    if not name:
-        # If it's a string, assume it's a parsed name that's already in the
-        # name map, find the leaf module name and add it
-        if isinstance(object, str):
+    if not module_name:
+        if object == 'os.PathLike':
```

Comment:
The changelog for https://github.com/pybind/pybind11/commit/5bcaaa0423c6757ca1c2738d0a54947dacdb03a1mentions that returned paths are exposed as `pathlib.Path`. I suspect it doesn't, based on the code there, but can you expand the test to check `std::filesystem::path` being used both as an argument and as a return value, just to be sure it doesn't behave differently for those?

Reply:
Great idea, thanks! Will do. 

I did notice that comment in pybind11, but didn't read it carefully enough.